### PR TITLE
feat: initial version of the extension pack

### PR DESCRIPTION
**Pull request by benoitf**
fixes https://github.com/containers/podman-desktop-internal/issues/169

### CODE REVIEW

File: .github/workflows/build-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+      - name: Login to ghcr.io
+        uses: redhat-actions/podman-login@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push To ghcr.io
+        id: push-to-quay
```
Comment (jeffmaury) [inline .github/workflows/build-publish.yaml:66]: That does not seem aligned with the name

File: Containerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+FROM scratch as builder
```
Comment (jeffmaury) [inline Containerfile:18]: What is the benefit of using the builder ?

Comment (benoitf) [reply Containerfile:18]: it reduces the size of the layers else you've like one layer per COPY entry

File: .github/workflows/build-publish.yaml (hunk 1 above)
Comment (benoitf) [inline .github/workflows/build-publish.yaml:66]:
```suggestion
        id: push-to-ghcr
```

**Comment by benoitf** [comment]
@jeffmaury applied changes

File: package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "redhat-pack",
+  "displayName": "Red Hat Extension Pack",
```
Comment (apupier) [inline package.json:3]: I think the displayName should be different. This extension pack is not covering *all* Red Hat technologies and extensions

Also the convention currently used by most extensions provided by Red Hat is followking the pattern `XXX by Red Hat`

Comment (apupier) [reply package.json:3]: or maybe... it is getting all things related to Red Hat in Podman Desktop...
I do not know in fact :-s

Comment (benoitf) [reply package.json:3]: it's something for Podman Desktop yes
I'm fine changing the wording but it needs to be short ;-)

'Extension Pack for Podman Desktop by Red Hat' is too long

Comment (benoitf) [reply package.json:3]: also it doesn't run in VS Code

File: .github/workflows/build-publish.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+jobs:
+  build:
+    name: Build image
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install qemu dependency
```
Comment (jeffmaury) [inline .github/workflows/build-publish.yaml:35]: Is it required ? Docs mentions only if RUN instructions ?

Comment (benoitf) [reply .github/workflows/build-publish.yaml:35]: else you can't do multi arch

**Comment by slemeur** [comment]
We may want to consider adding the extension https://github.com/redhat-developer/podman-desktop-redhat-account-ext if that's ready otherwise LGTM

**Comment by benoitf** [comment]
this extension is not yet publishing the new alpha versions but I'll add it there once it's available

File: package.json (hunk 3 above)
Comment (benoitf) [reply package.json:3]: keeping the name for now then.

display name can be changed easily later (and again it's not an extension that will go to a VS Code marketplace)

**Pull request merged**